kmm/kmm-feed/src/commonTest/kotlin/com/gchristov/newsfeed/kmmfeed/FeedViewModelTest.kt

```diff
@@ -10,15 +10,29 @@ import com.gchristov.newsfeed.kmmfeeddata.usecase.*
 import com.gchristov.newsfeed.kmmfeedtestfixtures.FakeFeedRepository
 import com.gchristov.newsfeed.kmmfeedtestfixtures.FeedCreator
 import com.gchristov.newsfeed.kmmposttestfixtures.FakePostRepository
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.*
 import kotlinx.datetime.Instant
 import kotlin.test.*
 
 @ExperimentalCoroutinesApi
 class FeedViewModelTest : CommonViewModelTestClass() {
 
+    @Test
+    fun searchQueryIsDebouncedOnExpectedInterval()  {
+        runTest { viewModel, _, _ ->
+            viewModel.onSearchTextChanged("te")
+            delay(100)
+            assertTrue { viewModel.state.value.searchQuery == "" }
```

Comment:
Ah this has to be updated to check for null with the new state change, which is why it’s failing.